libdnf/utils/locker.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "locker.hpp"
+
+#include "libdnf/common/exception.hpp"
+
+#include <errno.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+namespace libdnf::utils {
+
+Locker::Locker(const std::string & name) {
+    // unix domain socket path is declared as char[108]
+    libdnf_assert(name.size() <= 107, "Lock name is too large (maximum length 107 expected)");
+
+    // create a local unix socket
+    socket_fd = socket(AF_UNIX, SOCK_DGRAM, 0);
+    libdnf_assert(socket_fd > 0, "Cannot create a lock - socket creation failed.");
+
+    /* Set an abstract socket address */
+    struct sockaddr_un lock_address;
+    memset(&lock_address, 0, sizeof(lock_address));
+    lock_address.sun_family = AF_UNIX;
+    // first byte of abstract socket address is required to be \0
```

Comment:
The abstract socket namespace is a non portable Linux extension.
What to use something else? Pathname socket? Lock file?

Reply:
Right, it will work only on linux. What I liked on this solution is independence on any filesystem path. You just use arbitrary name for the lock. But in case we need compatibility across different operating systems, I would go with a lock file.